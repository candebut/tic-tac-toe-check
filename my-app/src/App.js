import React, { useState } from 'react';
import './App.css';

const App = () => {

  const [winner, setWinner] = useState(false)
  const [message, setMessage] = useState('')
  const tictactoe = [[0, 0, 0], [0, 0, 0], [0, 0, 0]];

  const isSolved = (board) => {
    if ((board[0][0] === 1 && board[0][1] === 1 && board[0][2] === 1) || //primera horizontal
      (board[1][0] === 1 && board[1][1] === 1 && board[1][2] === 1) || //segunda horizontal
      (board[2][0] === 1 && board[2][1] === 1 && board[2][2] === 1) || //tercera horizontal

      (board[0][0] === 1 && board[1][0] === 1 && board[2][0] === 1) || //primera vertical
      (board[0][1] === 1 && board[1][1] === 1 && board[2][1] === 1) || //segunda vertical
      (board[0][2] === 1 && board[1][2] === 1 && board[2][2] === 1) || //tercera vertical

      (board[0][0] === 1 && board[1][1] === 1 && board[2][2] === 1) || //primera diagonal
      (board[0][2] === 1 && board[1][1] === 1 && board[2][0] === 1))   //segunda diagonal
    {
      setWinner(true);
      setMessage("¡Ganó X!")
      return 1;
    }

    if ((board[0][0] === 2 && board[0][1] === 2 && board[0][2] === 2) || //primera horizontal
      (board[1][0] === 2 && board[1][1] === 2 && board[1][2] === 2) || //segunda horizontal
      (board[2][0] === 2 && board[2][1] === 2 && board[2][2] === 2) || //tercera horizontal

      (board[0][0] === 2 && board[1][0] === 2 && board[2][0] === 2) || //primera vertical
      (board[0][1] === 2 && board[1][1] === 2 && board[2][1] === 2) || //segunda vertical
      (board[0][2] === 2 && board[1][2] === 2 && board[2][2] === 2) || //tercera vertical

      (board[0][0] === 2 && board[1][1] === 2 && board[2][2] === 2) || //primera diagonal
      (board[0][2] === 2 && board[1][1] === 2 && board[2][0] === 2)) { //segunda diagonal
      setWinner(true);
      setMessage("¡Ganó 0!")
      return 2;
    }

    if ((board[0].includes(0)) || (board[1].includes(0)) || (board[2].includes(0))) {
      setMessage("¡El juego aún no terminó!")
      return -1;
    }
    if (winner === false && !board.includes(0)) {
      setMessage('¡Es un empate!');
      return 0;
    }
  }
  return (
    <>
      <div>{winner ? "¡Felicitaciones!" : "¡A jugar!"}</div>
      <button onClick={() => isSolved(tictactoe)}>Hacé click para ver el estado del juego</button>
      <div>{message}</div>
    </>
  )
}

export default App;
