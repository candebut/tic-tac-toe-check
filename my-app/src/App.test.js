import React from 'react';
import App from './App';
import { shallow, configure } from 'enzyme';
import { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App', () => {
  const container = shallow(<App />);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    mount(<App />);
  });

  describe('Button render correctly', () => {
    it('renders one button', () => {
      const wrapper = shallow(
        <App />
      );
      const element = wrapper.find('button');
      expect(element.length).toBe(1);
    });
  });
})
