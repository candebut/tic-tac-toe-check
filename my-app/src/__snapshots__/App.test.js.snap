// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `"<div>¡A jugar!</div><button>Hacé click para ver el estado del juego</button><div></div>"`;
